lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+        self.results['state'] = results
+
+        if self.tenant_id is None:
+            self.tenant_id = self.credentials['tenant']
+        if self.object_id is None:
+            self.object_id = self.credentials['client_id']
+
+        # return the results if your only gathering information
+        if self.check_mode:
+            return self.results
+
+        if changed:
+            if self.state == 'present':
+                # if you want to create or update a key vault
+                if not vault:
+                    # create new vault
+                    self.log('Creating vault {0}'.format(self.name))
+                    use_sku = Sku(self.sku)
+                    permission_preset = create_permissions(self)
+                    access_policy_lst = [AccessPolicyEntry(self.tenant_id, self.object_id, permission_preset, application_id=self.application_id)]
+                    vault_properties = VaultProperties(self.tenant_id,
+                                                       use_sku,
+                                                       access_policies=access_policy_lst,
+                                                       vault_uri=self.vault_uri,
+                                                       enabled_for_deployment=self.enabled_for_deployment,
+                                                       enabled_for_disk_encryption=self.enabled_for_disk_encryption,
+                                                       enabled_for_template_deployment=self.enabled_for_template_deployment)
+                    vault = VaultCreateOrUpdateParameters(self.location, vault_properties, tags=self.tags)
+                else:
+                    # update the vault
+                    vault = VaultCreateOrUpdateParameters(self.location, results['properties'], tags=results['tags'])
+
+                self.results['state'] = self.create_or_update_vault(vault)
+            elif self.state == 'absent':
+                # delete vault
+                self.delete_vault()
+                # the delete does not actually return anything. if no exception, then we'll assume
+                # it worked.
+                self.results['state']['status'] = 'Deleted'
+        return self.results
+
+    def create_or_update_vault(self, vault):
+        try:
+            # create or update the new Vault object we created
+            new_vault = self.keyvault_client.vaults.create_or_update(self.resource_group, self.name, vault)
+        except Exception as exc:
+            self.fail("Error creating or updating vault {0} - {1}".format(self.name, str(exc)))
+        return new_vault.as_dict()
+
+    def delete_vault(self):
+        try:
+            # delete the Vault
+            self.keyvault_client.vaults.delete(self.resource_group, self.name)
+        except Exception as exc:
+            self.fail("Error deleting vault {0} - {1}".format(self.name, str(exc)))
+        return None
+
+
+def create_permissions(self):
```

Comment:
Pass the self.permissions as parameter directly

Reply:
done